eric_sse/entities.py

```
@@ -1,55 +1,64 @@
+            return self.__queues[listener_id]
+        except KeyError:
+            raise InvalidListenerException(f"Invalid listener {listener_id}")
 
-    def on_message(self, msg: MessageContract) -> None:
-        """Event handler. It executes when a message is delivered to client"""
-        pass
+    def get_listener(self, listener_id: str) -> MessageQueueListener:
+        try:
+            return self.__listeners[listener_id]
+        except KeyError:
+            raise InvalidListenerException
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)